From d61682317410dafeb780ceed5a088cce56a414c8 Mon Sep 17 00:00:00 2001
From: Matt Roper <matthew.d.roper@intel.com>
Date: Thu, 23 Aug 2018 17:12:14 -0700
Subject: [PATCH] drm/i915: Add a little bit more debug output to MIPI-DSI
 readout

We'd like a little bit more information about how the BIOS setup our
MIPI panels before the driver started, especially when dual-link panels
are in use.

Signed-off-by: Matt Roper <matthew.d.roper@intel.com>
---
 drivers/gpu/drm/i915/vlv_dsi.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/vlv_dsi.c b/drivers/gpu/drm/i915/vlv_dsi.c
index 78fca98..034e55e7 100644
--- a/drivers/gpu/drm/i915/vlv_dsi.c
+++ b/drivers/gpu/drm/i915/vlv_dsi.c
@@ -1155,8 +1155,14 @@ static void bxt_dsi_get_pipe_config(struct intel_encoder *encoder,
 	 * encoder->get_hw_state() returns true.
 	 */
 	for_each_dsi_port(port, intel_dsi->ports) {
-		if (I915_READ(BXT_MIPI_PORT_CTRL(port)) & DPI_ENABLE)
+		u32 val = I915_READ(BXT_MIPI_PORT_CTRL(port));
+		if (val & DPI_ENABLE) {
+			DRM_DEBUG_KMS("DSI-%d: dual_link = %d, dl mode = %d\n",
+				      port,
+				      val & LANE_CONFIGURATION_DUAL_LINK_A,
+				      val & DUAL_LINK_MODE_MASK);
 			break;
+		}
 	}
 
 	fmt = I915_READ(MIPI_DSI_FUNC_PRG(port)) & VID_MODE_FORMAT_MASK;
-- 
1.9.1

